### ISSUE TITLE
Feature Request: support "Quick Look Preview extension" for macOS 10.13 and newer

### ISSUE THREAD

**Issue by Coeur** [transmission/transmission#3029]
Quick Look Preview extension is the replacement for the old Quick Look Plug-In (although, this one still works even in macOS 12 Monterey).

In Xcode, it looks like this:
![Capture d’écran 2022-04-30 à 20 07 35](https://user-images.githubusercontent.com/839992/166104900-c19319c3-dedb-4b22-a004-31a7a0f696f5.png)
It's presented in some WWDC videos like:
https://developer.apple.com/videos/play/wwdc2017/231 for macOS 10.13
https://developer.apple.com/videos/play/wwdc2018/237 for macOS 10.14
https://developer.apple.com/videos/play/wwdc2019/719 for macOS 10.15
The documentation is at:
https://developer.apple.com/documentation/quicklook/qlpreviewingcontroller?language=objc

**Comment by nevack** [comment]
I have taken a look into this issue and I can say it can be implemented on 10.15+.
I even have a prototype right now.

**Comment by nevack** [comment]
Old  QuickLook PlugIns (aka `.qlplugin`s) are dead with macOS 15!
https://developer.apple.com/documentation/macos-release-notes/macos-15-release-notes#Quick-Look

**Comment by nevack** [comment]
I have a working prototype of Data-based Quick Look Preview appex, that reuses all html based rendering logic from legacy plugin.

<img width="1200" alt="image" src="https://github.com/user-attachments/assets/01383b1b-b8f1-4a7c-b7b7-3fea2348ea26">

Assigning this to myself as I will try to publish finished version this week.

The only downside - Data-based Quick Look Preview appex are supported only on macOS 12+
I researched a bit, what people do for 10.14-11.0 and the only solution I found - creating WebKit view and serving content in here. [See here for inspiration](https://github.com/sbarex/SourceCodeSyntaxHighlight/blob/master/QLExtension/PreviewViewController.swift)

**Comment by Coeur** [comment]
>The only downside - Data-based Quick Look Preview appex are supported only on macOS 12+

Can we keep both the `Quick Look Plug-In` and the `Quick Look Preview appex`?
Otherwise, as a non-essential feature for torrenting (it's literally useless for downloading/seeding), I'd be exceptionally OK to drop **Quick Look** support for macOS 11 and older.

**Comment by 2600box** [comment]
> Assigning this to myself as I will try to publish finished version this week.

Amazing. Will be looking out for this!

**Comment by nevack** [comment]
> Can we keep both the `Quick Look Plug-In` and the `Quick Look Preview appex`?

Indeed, I made new extension to be used on macOS 12+, so legacy plugin is used on previous versions as expected.

Testing this required some burdensome setup, but here it is:
<img width="1200" alt="Screenshot 2024-11-10 at 21 34 54" src="https://github.com/user-attachments/assets/7fcc547c-768c-4178-860f-bfd7504c4352">

**Comment by drinkius** [comment]
What is the status of this new appex? Is there a way to install it?
